otter/integration/lib/autoscale.py

```diff
@@ -72,6 +84,74 @@ def decide(resp):
             .addCallback(decide)
         )
 
+    def wait_for_N_servers(
+        self, rcs, servers_desired, period=10, timeout=600, clock=None
+    ):
+        """Waits for the desired number of servers.
+
+        :param TestResources rcs: The integration test resources instance.
+            This provides useful information to complete the request, like
+            which endpoint to use to make the API request.
+        :param int servers_desired: The number of servers to wait for.
+            It must be greater than or equal to one.
+        :param int period: The number of seconds between poll attempts.
+            If unspecified, defaults to 10 seconds.
+        :param int timeout: The number of seconds to wait before giving up.
+            Defaults to 600 seconds (10 minutes).
+        :param twisted.internet.interfaces.IReactorTime clock: If provided,
+            the clock to use for scheduling things.  Defaults to `reactor`
+            if not specified.
+
+        :return: If the operation succeeds, the same instance of TestResources.
+            Otherwise, an exception is raised.
+        """
+
+        clock = clock or reactor
+
+        class Looper(object):
+            """Inner-class to support looping until either of two conditions
+            holds: (1) we see at least `servers_desired` servers in existence,
+            or (2) a timeout is reached.
+            """
+
+            def __init__(self, sg):
+                self.elapsed_time = 0
+                self.sg = sg
+                # To be filled in later.
+                self.loopingCall = None
+
+            def loop(self):
+                if self.elapsed_time < timeout:
+                    self.elapsed_time += period
+                    return self.check_servers()
+                else:
+                    raise TimeoutError(
+                        "Spent %ds, polling every %ds, timeout." % (
+                            self.elapsed_time, period
+                        )
+                    )
+
+            def ensure_200_result(self, state_results):
+                if state_results[0] != 200:
+                    raise Exception("Scaling group not found.")
```

Comment:
200 and 404 are listed as "valid results" for the get_scaling_group_details call.  Any other result will throw an exception.  In this case, I'm also discarding the 404 valid result, since it's not valid for our needs here.